### ISSUE TITLE
F.softmax missing dim parameter causes incorrect behavior

### ISSUE THREAD

**Issue by xiang1ian** [hfslyc/AdvSemiSeg#73]
Line 345 : D_out = interp(model_D(F.softmax(pred)))
Here and in several other places, the dim parameter is not specified. This defaults to dim=-1, applying softmax along the width dimension. However, for semantic segmentation with pred shape (B, C, H, W), softmax should be applied along the channel dimension (dim=1) to produce valid per-pixel class probabilities.

This is both a deprecation warning in recent PyTorch versions and a functional bug that breaks the adversarial training objective.
